### ISSUE TITLE
Specify custom generated go filenames and cli command parameter name

### ISSUE THREAD

**Issue by hectorj2f** [goadesign/goa#1035]
We faced an issue on which the resource name have to differ from the generated filename and the name of the cli command argument.

The resource name is used as the filename prefix for the golang file, as defined here: (https://github.com/goadesign/goa/blob/master/goagen/gen_main/generator.go#L112). Moreover, the cli command argument seems to use the resource name as well:(https://sourcegraph.com/github.com/goadesign/goa@450e184b114536525efffa64e2b31733ee689d97/-/blob/goagen/gen_client/cli_generator.go#L837:27-837:29).

For instance, we want to define a command where the resource name can differ for the cli argument name, e.g:

```
var _ = a.Resource("obj_iterations", func() {
	a.Parent("obj")
	a.Action("list", func() {
		a.Routing(
			a.GET("obj"),
		)
		a.Description("List objs.")
		a.Response(d.OK, func() {
			a.Media(objList)
		})
		a.Response(d.InternalServerError, ...)
	})
```
We want to get a filename that uses underscores `obj_iterations.go` but that looks like that in our cli:
`cli create obj-iterations`. However we get the same cli parameter `cli create obj_iterations` which isn't what we want. Moreover, the use of underscores in cli apps isn't correct.

Is there a way of customizing this name for the cobra command ?

**Comment by hectorj2f** [comment]
An easy solution could be to replace the `underscores` by `dashes` when using the `resource.Name` as cmd argument.

**Comment by raphael** [comment]
Thanks for the detailed report! Replacing underscores with dashes for the CLI flags makes sense to me. It should probably done for all the flags though - not just resource names. Would you care to do the honor with a PR?

**Comment by hectorj2f** [comment]
@raphael I made a PR with some changes, please have a look at it :).

**Comment by goldcaddy77** [comment]
This can be closed since #1039 is merged, right?

**Comment by raphael** [comment]
Yup, thank you!
